src/nl/rubensten/texifyidea/intentions/LatexLabelDefiningNewCommand.kt

```diff
@@ -0,0 +1,112 @@
+package nl.rubensten.texifyidea.intentions
+
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiFile
+import nl.rubensten.texifyidea.psi.LatexCommands
+import nl.rubensten.texifyidea.settings.LabelingCommandInformation
+import nl.rubensten.texifyidea.settings.TexifySettings
+import nl.rubensten.texifyidea.settings.labeldefiningcommands.EditLabelDefiningCommand
+import nl.rubensten.texifyidea.util.*
+
+open class LatexLabelDefiningNewCommand : TexifyIntentionBase("Add label defining command to list") {
+    private val settings = TexifySettings.getInstance()
+
+    override fun isAvailable(project: Project, editor: Editor?, file: PsiFile?): Boolean {
+        if (editor == null || file == null || !file.isLatexFile()) {
+            return false
+        }
+
+        // check if selected part is \label
+        val element = file.findElementAt(editor.caretModel.offset) ?: return false
+        var selected = element as? LatexCommands ?: element.parentOfType(LatexCommands::class) ?: return false
+
+        // get parent element
+        val parentElement = selected.parent
+        val parent = parentElement as? LatexCommands
+                ?: parentElement.parentOfType(LatexCommands::class)
```

Comment:
can be a oneliner.

Reply:
fixed in commit 4fc09f4f03cfda54b13164fb3b1d6e31c694f878

Reply:
fixed in commit 4fc09f4f03cfda54b13164fb3b1d6e31c694f878